agent/reconciler/host_reconciler.go

```diff
@@ -156,6 +169,29 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 	return ctrl.Result{}, nil
 }
 
+func (r *HostReconciler) installerController(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) (ctrl.Result, error) {
+	logger := ctrl.LoggerFrom(ctx)
+	installScript, uninstallScript, err := r.getInstallationScript(ctx, byoHost.Spec.InstallationSecret.Name, byoHost.Spec.InstallationSecret.Namespace)
+	if err != nil {
+		logger.Error(err, "error getting installation script")
+		r.Recorder.Eventf(byoHost, corev1.EventTypeWarning, "ReadInstallationSecretFailed", "installation secret %s not found", byoHost.Spec.InstallationSecret.Name)
+		return ctrl.Result{}, err
+	}
+	byoHost.Spec.UninstallationScript = &uninstallScript
+	installScript, err = r.parseInstallationScript(ctx, installScript)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	logger.Info("executing install script")
+	err = r.executeScript(ctx, installScript)
```

Comment:
You can reuse the `RunCmd` helper - https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/b09a31d78eb0b2e824f3f98dad96f2ead0f638a8/agent/cloudinit/cmd_runner.go#L21-L26

Reply:
`HostReconciler` already have `CmdRunner`, we can use it to execute the script.

Reply:
But we are executing with context `exec.CommandContext`. Should I change it?

Reply:
This is a bash script so we use our own executor.

Reply:
TIL [CommandContext](https://pkg.go.dev/os/exec#CommandContext), thanks.
Maybe make the `CmdRunner` use `CommandContext` as well? Or an option to run using the `CommandContext`?

Reply:
we can have seperate PR for this

Reply:
This does not look like a major change and is well within the scope of this PR. So I'd prefer for it to be part of this PR itself.
Am I missing something?

Reply:
We need to execute the script using `bash` (as `sh` don't support `set -o pipefail`).

I think it will be ok to change in existing [RunCmd()](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/b09a31d78eb0b2e824f3f98dad96f2ead0f638a8/agent/cloudinit/cmd_runner.go#L22) function:
`command := exec.Command("/bin/sh", "-c", cmd)` to 
`command := exec.CommandContext("/bin/bash", "-c", cmd)`
WDTY @anusha94 ?

Reply:
Yes, feel free to make changes to existing helper methods.
As long as tests are passing, we should be good!